clikit/src/main/java/org/adamalang/clikit/exceptions/XMLFormatException.java

```diff
@@ -0,0 +1,38 @@
+package org.adamalang.clikit.exceptions;
+
+import java.util.ArrayList;
+
+public class XMLFormatException extends Exception {
```

Comment:
So, I understand this as an ErrorBuilder. However, the stack track may not be the most useful way. However, for a CLI tool, this is fine.

However, something that is worth investigating is if the org.w3c.dom provide the ability to illustrate the position of an element in a document. A better error message would be "The '$tag' node at line 42 char 27 is missing a name attribute".

Reply:
I have added error messages that show element position in document, but I might make it so that it throws the first error. This is solely because it may be hard to read if there are a lot of lines, but I can probably format it to be more readable.